pkg/agent/proxy/proxier.go

```diff
@@ -534,14 +534,17 @@ func (p *proxier) installServices() {
 			}
 
 			// Install ClusterIP flows for the Service.
-			groupID := p.groupCounter.AllocateIfNotExist(svcPortName, internalPolicyLocal)
+			groupID, exists := p.groupCounter.Get(svcPortName, internalPolicyLocal)
```

Comment:
Guess let's make the PR focus on not watching endpoints when endpointslice is enabled, and leave the bugfix and other improvements in a separate PR.

Reply:
Reverted